zhaquirks/tuya/valve.py

```diff
@@ -0,0 +1,200 @@
+from zigpy.zcl.clusters.hvac import Thermostat
+from zigpy.zcl import foundation
+import zigpy.types as t
+
+from .. import Bus, LocalDataCluster
+from ..const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+from . import TuyaManufClusterAttributes
+
+# info from https://github.com/Koenkk/zigbee-herdsman-converters/pull/1120 and
+# https://github.com/dresden-elektronik/deconz-rest-plugin/blob/master/tuya.cpp
+# they do not agree for 0x0112 and 0x0114: the former things the attribute is to
+# en/dis-able the detection and reporting, whereas the latter thinks the attrib
+# contains the state report itself.
+SITERWELL_CHILD_LOCK_ATTR = 0x0107  # [0] unlocked [1] child-locked
+SITERWELL_WINDOW_OPEN_ATTR = 0x0112  # [0] window closed [1] window open detected
+SITERWELL_VALVE_STATE_ATTR = 0x0114  # [0] valve closed [1] valve open
+SITERWELL_TARGET_TEMP_ATTR = 0x0202  # [0,0,0,210] target room temp (decidegree)
+SITERWELL_TEMPERATURE_ATTR = 0x0203  # [0,0,0,200] current room temp (decidegree)
+SITERWELL_BATTERY_ATTR = 0x0215  # [0,0,0,98] battery charge
+SITERWELL_MODE_ATTR = 0x0404  # [0] off [1] auto [2] heat
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TuyaManufClusterSiterwellGS361(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Siterwell GS361 device."""
+
+    manufacturer_attributes = {
+        SITERWELL_CHILD_LOCK_ATTR: ("child_lock", t.uint8_t),
+        SITERWELL_WINDOW_OPEN_ATTR: ("window_open", t.uint8_t),
+        SITERWELL_VALVE_STATE_ATTR: ("valve_state", t.uint8_t),
+        SITERWELL_TARGET_TEMP_ATTR: ("target_temperature", t.uint32_t),
+        SITERWELL_TEMPERATURE_ATTR: ("temperature", t.uint32_t),
+        SITERWELL_BATTERY_ATTR: ("battery", t.uint32_t),
+        SITERWELL_MODE_ATTR: ("mode", t.uint8_t),
+    }
+
+    TEMPERATURE_ATTRS = {
+        SITERWELL_TEMPERATURE_ATTR: "local_temp",
+        SITERWELL_TARGET_TEMP_ATTR: "occupied_heating_setpoint",
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid in self.TEMPERATURE_ATTRS:
+            self.endpoint.device.thermostat_bus.listener_event(
+                "temperature_change",
+                self.TEMPERATURE_ATTRS[attrid],
+                value * 10,  # decidegree to centidegree
+            )
+        elif attrid == SITERWELL_MODE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event("mode_change", value)
+        elif attrid == SITERWELL_VALVE_STATE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event("valve_state", value)
+
+
+class SiterwellGS361Thermostat(LocalDataCluster, Thermostat):
+    """Thermostat cluster for Siterwell GS361 device."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.switch_bus.add_listener(self)
```

Comment:
I think it's more self.endpoint.device.thermostat_bus.add_listener(self)

Reply:
And again, bad copy-paste.